src/components/userFeedback/UserFeedback.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { extend } from 'underscore';
+
+export interface IUserFeedbackProps {
+  feedbackText: string;
+  state: string;
+  extraClasses?: string;
+  displayOnShow?: string;
+}
+
+export interface IUserFeedbackStyle {
+  displayClass: string;
+  textColorClass: string;
+  extraClasses?: string;
+}
+
+export const UserFeedbackState = {
+  VALID: 'VALID',
+  WARNING: 'WARNING',
+  ERROR: 'ERROR',
+};
+
+export const TextColorClasses = {
+  default: 'text-dark-grey',
+  error: 'text-red'
+};
+
+export class UserFeedback extends React.Component<IUserFeedbackProps, any> {
+
+  private getDefaultUserFeedbackStyle(): IUserFeedbackStyle {
+    return {
+      displayClass: 'hidden',
+      textColorClass: TextColorClasses.default,
+      extraClasses: this.props.extraClasses ? ' ' + this.props.extraClasses : ''
+    };
+  }
+
+  private adjustStyle(state: string): IUserFeedbackStyle {
+    switch (state) {
+      case UserFeedbackState.VALID:
+        return this.getDefaultUserFeedbackStyle();
+
+      case UserFeedbackState.WARNING:
+        return extend(this.getDefaultUserFeedbackStyle(), { displayClass: this.props.displayOnShow || 'block' });
+
```

Comment:
Break is not needed. Since the function only contains a switch statement and does not need to continue to evaluate the function after the switch statement, break is not needed here. The function returns as soon as it finds a case match.